frontend/src/components/body/hooks/useQuery.ts

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect, useCallback } from "react";
```

Comment:
1. Set initial state properly
    2. Use generics in useState hook to type data and fix type of error. Currently type of data and error is any.
    3. api <> uri (means rename api to url) and refetchDataFromServer <> refetch.
    4. Prefer try catch syntax over .then.
    5. Store backend api url in .env file and read it from there. -> http://localhost:5000/api
    6. Need abort handling in case of manual refetch.
    7. Issue related to race condition -> calling refetch just after initial mount and calling refetch more than once.